packages/yourloops/test/integration/assert/caregivers-list.assert.ts

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { screen, within } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { mockCaregiverUser } from '../mock/direct-share.api.mock'
+import NotificationApi from '../../../lib/notifications/notification.api'
+import { NotificationType } from '../../../lib/notifications/models/enums/notification-type.enum'
+import { type Notification } from '../../../lib/notifications/models/notification.model'
+import DirectShareApi, { PATIENT_CANNOT_BE_ADDED_AS_CAREGIVER_ERROR_MESSAGE } from '../../../lib/share/direct-share.api'
+import { patient1Id } from '../data/patient.api.data'
+
+export const checkCaregiversListLayout = async () => {
+  expect(await screen.findByTestId('patient-caregivers-list')).toBeVisible()
+
+  const addCaregiverButton = screen.getByRole('button', { name: 'Add caregiver' })
+  expect(addCaregiverButton).toBeVisible()
+  expect(addCaregiverButton).toBeEnabled()
+
+  const caregiversTable = screen.getByTestId('patient-caregivers-list')
+  expect(caregiversTable).toHaveTextContent('Last nameFirst nameEmail')
+
+  const caregiverRow = within(caregiversTable).getByTestId(`patient-caregivers-table-row-${mockCaregiverUser.username}`)
+  expect(caregiverRow).toBeVisible()
+  expect(caregiverRow).toHaveTextContent('UserCaregivercaregiver@mail.com')
+  expect(within(caregiverRow).getByTestId(`patient-caregivers-table-row-${mockCaregiverUser.username}-button-remove`)).toBeVisible()
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**🤓 nitpick:** ‏It is best to get buttons by role and aria label (when the button has no text). Doing it this way also ensure that the button has indeed an aria label :slightly_smiling_face: